src/spartan_handler_fsm.erl

```diff
@@ -149,15 +130,26 @@ execute(timeout, State = #state{data = Data}) ->
             spartan_metrics:update([?APP, no_upstreams_available], 1, ?SPIRAL),
             reply_fail(State1),
             {stop, normal, State};
+        [Upstream] ->
+            StartTimestamp = os:timestamp(),
```

Comment:
I thought process creation is cheap. Is it really giving us any benefit? If not, then it would be good to keep the logic same irrespective of number of upstreams